app/src/main/java/com/vishaltelangre/nerdcalci/core/DateStringParser.kt

```
@@ -0,0 +1,164 @@
+     *   - If that date is within the past 6 months, use current year.
+     *   - If that date is more than 6 months in the past, use next year.
+     *   - If that date is in the future (any amount), use current year.
+     *
+     * This matches general Soulver behavior: "the occurrence closest to today;
+     * prefer future when equidistant".
+     */
+    private fun buildDate(year: Int?, month: Int, day: Int, original: String): DateTimeResult {
+        val resolvedYear = year ?: inferYear(month, day)
+        return try {
+            DateTimeResult.Date(LocalDate.of(resolvedYear, month, day))
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)